lectures/zksnark.tex

```diff
@@ -0,0 +1,812 @@
+            1 & 0 & -1 & 0 & 0 & 0 & 0 
+        \end{pmatrix}
+    \end{equation*}
+\end{example}
+
+The columns of these matrices represent the mappings from constraint number $i$ to the corresponding
+coefficient of the $j$ element in the witness vector.
+
+\begin{example}
+    Considering the witness from the previos examples: 
+    \begin{equation*}
+         w = (1, r, x_1, x_2, x_3, \mathsf{mult}, \mathsf{selectMult})
+    \end{equation*}
+    For element $x_1$ we are interesting in the third columns of the $A$, $B$ and $C$ matrices, as
+    it's placed on the third position in the witness vector, so $j = 3$.\\
+    For matrix $A$:
+    \begin{center}
+        \begin{tikzpicture}
+            % Node to contain the pmatrix
+            \node (A) at (0,0) {$
+                \begin{pmatrix}
+                    0 & 0 & 1 & 0 & 0 & 0 & 0 \\
+                    0 & 0 & 0 & 1 & 0 & 0 & 0 \\
+                    0 & 0 & 1 & 0 & 0 & 0 & 0 \\
+                    1 & 0 & -1 & 0 & 0 & 0 & 0 \\
+                \end{pmatrix}
+            $};
+        
+            % Draw the red rectangle around the third column
+            \draw[red, thick] 
+                ([xshift=-77pt,yshift=-2pt]A.north east) -- ++(0,-2.25) -- 
+                ++(-0.75,0) -- ++(0,2.25) -- cycle;
+        
+        \end{tikzpicture}
+    \end{center}
+
+    Thus, for constraint number 4 ($i = 4$) the coefficient of $x_1$ is $-1$:
+    \begin{center}
+        \begin{tikzpicture}
+            % Node to contain the pmatrix
+            \node (A) at (0,0) {$
+                \begin{pmatrix}
+                    0 & 0 & 1 & 0 & 0 & 0 & 0 \\
+                    0 & 0 & 0 & 1 & 0 & 0 & 0 \\
+                    0 & 0 & 1 & 0 & 0 & 0 & 0 \\
+                    1 & 0 & -1 & 0 & 0 & 0 & 0 \\
+                \end{pmatrix}
+            $};
+        
+            % Draw the red rectangle around the third column
+            \draw[red, thick] 
+                ([xshift=-77pt,yshift=-2pt]A.north east) -- ++(0,-2.25) -- 
+                ++(-0.75,0) -- ++(0,2.25) -- cycle;
+
+            \draw[OliveGreen, thick] 
+                ([xshift=-10pt,yshift=-48pt]A.north east) -- ++(-4.3,0) -- 
+                ++(0,-0.63) -- ++(4.3,0) -- cycle;
+        
+        \end{tikzpicture}
+    \end{center}
+\end{example}
+
+As we know from the previous chapters, such a mapping in math can be built using polynomial 
+interpolation.
+
+\begin{remark}
+    As a remainder the Lagrange interpolation polynomial for a given set of points 
+    $\{(x_0,y_0),(x_1,y_1),\dots,(x_n,y_n)\} \subset \mathbb{F} \times \mathbb{F}$
+    can be built with the following formula:
+    \begin{equation*}
+        L(x) = \sum_{i=0}^{n} y_i \ell_i(x), \quad \ell_i(x) = \prod_{j=0, j \neq i}^{n} \frac{x-x_j}{x_i-x_j}.
+    \end{equation*}  
+\end{remark}
+
+For a given column $j \in \{1, 2, \dots, n\}$ in a matrix $A$ the set of points that define the
+variable polynomial $a_j(x)$ can be defined as follows:
```

Comment:
Use some other letter for representing polynomials (since you used $a$ multiple times before). Maybe capital $A_j(x)$ or $P^A_j(x)$ or $L_j(x)$ (corresponding to **left** input writes to the gates)?